pkg/api/pluginproxy/pluginproxy.go

```diff
@@ -80,15 +91,35 @@ func NewApiPluginProxy(ctx *models.ReqContext, proxyPath string, route *plugins.
 		}
 
 		if err := setBodyContent(req, route, data); err != nil {
-			logger.Error("Failed to set plugin route body content", "error", err)
+			appProxyLogger.Error("Failed to set plugin route body content", "error", err)
 		}
 	}
 
-	return &httputil.ReverseProxy{Director: director, ModifyResponse: modifyResponse}
+	logAppPluginProxyRequest(appID, cfg, ctx)
+
+	return proxyutil.NewReverseProxy(appProxyLogger, director)
 }
 
-func modifyResponse(resp *http.Response) error {
-	proxyutil.SetProxyResponseHeaders(resp.Header)
+func logAppPluginProxyRequest(appID string, cfg *setting.Cfg, c *models.ReqContext) {
+	if !cfg.DataProxyLogging {
+		return
+	}
+
+	var body string
+	if c.Req.Body != nil {
+		buffer, err := ioutil.ReadAll(c.Req.Body)
+		if err == nil {
+			c.Req.Body = ioutil.NopCloser(bytes.NewBuffer(buffer))
+			body = string(buffer)
+		}
+	}
 
-	return nil
+	logger.Info("Proxying incoming request",
+		"userid", c.UserId,
+		"orgid", c.OrgId,
+		"username", c.Login,
+		"app", appID,
+		"uri", c.Req.RequestURI,
+		"method", c.Req.Method,
+		"body", body)
```

Comment:
I don't know if we used to log the body before but if not, what if it contains sensitive information? Is that possible? In which cases `DataProxyLogging` is enabled? Would debug level be more appropriate?

Reply:
We've done that since at least [5 years](https://github.com/grafana/grafana/commit/63d6ab476a9a767ff5bc7d4e093b5ec303c70d5d) so I don't see a problem with it. Just added it to the plugin proxy now as well. It's only the request body posted to Grafana HTTP API.

> Would debug level be more appropriate?

I think no, because you have to actively enable data proxy logging for it to be logged.